browser/main/lib/dataApi/attachmentManagement.js

```diff
@@ -253,23 +261,57 @@ function generateAttachmentMarkdown (fileName, path, showPreview) {
  * @param {Event} dropEvent DropEvent
  */
 function handleAttachmentDrop (codeEditor, storageKey, noteKey, dropEvent) {
-  const file = dropEvent.dataTransfer.files[0]
-  const filePath = file.path
-  const originalFileName = path.basename(filePath)
-  const fileType = file['type']
-  const isImage = fileType.startsWith('image')
-  let promise
-  if (isImage) {
-    promise = fixRotate(file).then(base64data => {
-      return copyAttachment({type: 'base64', data: base64data, sourceFilePath: filePath}, storageKey, noteKey)
+  if (dropEvent.dataTransfer.files.length > 0) {
+    const file = dropEvent.dataTransfer.files[0]
+    const filePath = file.path
+    const originalFileName = path.basename(filePath)
+    const fileType = file['type']
+    const isImage = fileType.startsWith('image')
+    let promise
+    if (isImage) {
+      console.log(file)
+      promise = fixRotate(file).then(base64data => {
+        return copyAttachment({type: 'base64', data: base64data, sourceFilePath: filePath}, storageKey, noteKey)
+      })
+    } else {
+      promise = copyAttachment(filePath, storageKey, noteKey)
+    }
+    promise.then((fileName) => {
+      const imageMd = generateAttachmentMarkdown(originalFileName, path.join(STORAGE_FOLDER_PLACEHOLDER, noteKey, fileName), isImage)
+      codeEditor.insertAttachmentMd(imageMd)
     })
   } else {
-    promise = copyAttachment(filePath, storageKey, noteKey)
+    for (let i = 0; i < dropEvent.dataTransfer.items.length; i++) {
+      const item = dropEvent.dataTransfer.items[i]
+
+      if (item.type === 'text/html') {
+        const html = dropEvent.dataTransfer.getData('text/html')
+
+        const match = /<img[^>]*src="([^"]+)"/.exec(html)
+        if (match) {
+          const imageURL = match[1]
+
+          getImage(imageURL)
+            .then(image => {
+              const canvas = document.createElement('canvas')
+              const context = canvas.getContext('2d')
+              canvas.width = image.width
+              canvas.height = image.height
+              context.drawImage(image, 0, 0)
+
+              return copyAttachment({type: 'base64', data: canvas.toDataURL(), sourceFilePath: imageURL}, storageKey, noteKey)
+            })
+            .then(fileName => {
```

Comment:
At first glance, it looks like this is duplicate code with some lines above.. is there a way to implement your changes by minimizing the duplicate code? Or is this the best solution?